packages/openapi-sampler/src/samplers/StringSampler.ts

```diff
@@ -55,30 +55,60 @@ export class StringSampler implements Sampler {
     const randExp = new RandExp(pattern);
 
     if (min) {
-      // ADHOC: make a probe for regex using min quantifier value
-      // e.g. ^[a]+[b]+$ expect 'ab', ^[a-z]*$ expect ''
+      return this.sampleMinLength(randExp, min, max);
+    }
+
+    randExp.max = max ?? randExp.max;
+    randExp.randInt = (a, b) => Math.floor((a + b) / 2);
+
+    const result = randExp.gen();
+
+    return !!max && result.length > max && this.hasInfiniteQuantifier(pattern)
+      ? this.sampleInfiniteQuantifier(randExp, max)
+      : result;
+  }
+
+  private hasInfiniteQuantifier(pattern: string | RegExp) {
+    const pat = pattern.toString();
```

Comment:
`RegExp#toString()` and `pattern` keyword is not compatible 

```suggestion
    const pat = pattern.source;
```